### DISCUSSION TITLE
DOIs and citations for popular georust repositories

### DISCUSSION THREAD

**Discussion by urschrei** [georust/geo#799] [Ideas]
As some of Georust's projects have recently become more popular in academic settings, I'd like to assign DOIs and citation suggestions – which would be displayed in the project README and possibly docs – to them. This is free (using e.g. Zenodo), and low- to no-maintenance (a sample citation file looks like https://github.com/urschrei/simplification/blob/master/CITATION.cff)

I'd like to do this because: as an academic, citation of _my_ work matters to me, both for professional reasons and as a courtesy. For those of you who are not academics there's probably little to no professional benefit, but your work should still be recognised.

What do people think? I've done this for some of my own projects (e.g. https://github.com/urschrei/simplification), and I'm happy to do this work myself. In keeping with the collaborative nature of georust, authors would be named simply as "The Georust authors" for now, though there may be some exceptions here (rstar?).

**Comment by lnicola** [comment]
I think this is somewhat unusual, but only because it's not more common. But I don't see any downsides to it, so :+1:.

**Comment by dabreegster** [comment]
+1 to doing this. @robinlovelace and I just submitted a paper acknowledging georust at the end, but even better if we could cite it

**Comment by michaelkirk** [comment]
I'm not in academia, but this seems great. Do you know of any other projects that do this that we could learn from?

I have some technical questions about how to write and maintain the metadata.

For example, when looking at https://github.com/urschrei/simplification/blob/master/CITATION.cff

> message: "If you use this software, please cite it using these metadata."

I understood this proposal to be directed at people publishing research - not just any old usage. E.g. if rgis utilizes the `simplification` crate in it's application, should it somehow include this citation somewhere? I'd guess not, but want to make sure I'm understanding. 

> version: "0.5.21"

I guess this is something that'd need to be updated as part of the release process?

Having some guidelines (maybe in https://github.com/georust/meta) would be good.

**Comment by urschrei** [comment]
> Do you know of any other projects that do this that we could learn from?

1. The `pandas` library has citation guidelines: https://pandas.pydata.org/about/citing.html
2. The `numpy` library, too: https://numpy.org/citing-numpy/
3. The APA guidelines on when you should cite software: https://blog.apastyle.org/apastyle/2015/01/how-to-cite-software-in-apa-style.html

> I understood this proposal to be directed at people publishing research - not just any old usage. E.g. if rgis utilizes the simplification crate in it's application, should it somehow include this citation somewhere? I'd guess not, but want to make sure I'm understanding.

Correct – I'm referring only to usage in academic research.

> I guess this is something that'd need to be updated as part of the release process?

Yep, the version should be kept up to date, and I'm happy to add guidelines to meta.

It's possible that the georust project will form the _basis_ of an academic publication at some point (e.g. https://www.nature.com/articles/s41586-020-2649-2), at which point we would update the citation metadata to point to it and its new DOI.

**Reply by lnicola** [reply]
> Yep, the version should be kept up to date, and I'm happy to add guidelines to meta.

The easiest way to do this is probably adding a test that searches that file for `env!("CARGO_PKG_VERSION")`.
